RereleaseTimer/index.js

```diff
@@ -1,203 +1,100 @@
+
+    context.log(`Processing rule for ${app} (${sourceGroup} -> ${destinationGroup})...`);
+    const releases = await appCenterApi.getRecentReleases(owner, app);
 
-function GetLatest(releases, group) {
-    for(z = 0; z < releases.length; z++) {
-        if(IsInGroup(releases[z], group)) {
-            return releases[z];
-        }
+    // Getting the latest release from the source group:
+    const release = utils.getLatestRelease(releases, sourceGroup);
+    if (!release) {
+        context.log("No releases available in source group.");
+        return;
     }
-}
 
-function IsInGroup(release, group) {
-    if(release.distribution_groups) {
-        for(i = 0; i < release.distribution_groups.length; i++) {
-            if(release.distribution_groups[i].name == group)
-                return true;
-        }
+    // If it has been already released to the destination group, stop the execution:
+    if (utils.isInGroup(release, destinationGroup)) {
+        context.log(`Latest release (${release.short_version}) has already been distributed to the destination group.`);
+        return;
     }
-    return false;
-}
 
-function FindOne(endpoint, token, owner, app) {
-    var options = BuildUrl(endpoint, token, owner, app);
-    return request(options)
-    .then(result => {
-        result = JSON.parse(result);
-        if (result) {
-            return result;
-        }
-    })
-}
+    context.log(`Checking stats for version ${release.short_version} (${release.id})...`);
+    const stats = await getStats(release, owner, app);
+    let [crashes, sessions, installations] = [...stats];
+    context.log(`Crashes Detected: ${crashes}`);
+    context.log(`Sessions (30sec-30min): ${sessions}`);
+    context.log(`Total Installations: ${installations}`);
+
+    // Proceed with the release only if 
+    // - the amount of crashes since the release has been made 
+    //   does not exceed the maximum amount specified in config file;
+    // - the amount of installations since the release has been made 
+    //   is higher than the specified in config file;
+    // - the amount of event sessions since the release has been made 
+    //   is higher than the specified in config file;
+    if (!(crashes <= maxCrashes && installations >= minInstallations && sessions >= minSessions)) {
+        context.log(`Nothing to perform.`);
+        return;
+    }
 
-function GetDestination(token, owner, app, rule) {
-    switch(rule.type) {
-        case "store":
-            return FindOne(`/distribution_stores/${rule.destination}`, token, owner, app);
-        default:
-            return FindOne(`/distribution_groups/${rule.destination}`, token, owner, app);
+    context.log(`Re-releasing latest version...`);
+    const group = await appCenterApi.getDestinationGroup(owner, app, rule);
+    if (!group) {
+        throw new Error("Could not lookup destination group for re-release.");
+    }
+    const newRelease = await appCenterApi.getRelease(owner, app, release.id);
```

Comment:
I believe we can wrap up this line and `if` below to the function.

Reply:
I mean the logic here is about creating an release if I get it right, so we can wrap up these locs:
```js
const newRelease = await appCenterApi.getRelease(owner, app, release.id);
if (!newRelease) {
        return;
    } else {
        const patchRelease = {
            destinations: [{ id: group.id, name: group.name }],
            mandatory_update: newRelease.mandatory_update,
            release_notes: newRelease.release_notes
        };
        return await appCenterApi.makeRelease(owner, app, newRelease.id, patchRelease);
     }
```